packages/api/endpoints/reconciliation-reports.js

```diff
@@ -1,3 +1,5 @@
+//@ts-check
```

Comment:
I had to add in some type checking and guards, defaults, etc to satisfy some of the ts-check complaints. If there are better ways of handling some of that, I'm open to suggestions. I just followed some style guides and tried to make choices that made sense.